### PR TITLE
Improve build system

### PR DESCRIPTION
**Pull request by megabyde**
* Added dedicated CMake target with the `main` function and Makefile target to run it
* Added automatic discovery of test source files

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -8,11 +8,17 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
...
 find_package(Catch2 REQUIRED)
 
-add_executable(test_something test_something.cpp)
-add_test(test_something test_something)
-target_link_libraries(test_something PRIVATE Catch2::Catch2)
+add_executable(main main.cpp)
 
-add_custom_target(  
+file(GLOB tests "${PROJECT_SOURCE_DIR}/test_*.cpp")
+foreach(test ${tests})
```
Comment (megabyde) [inline CMakeLists.txt:14]: I don't remember if this follows the naming we used to use: variable for test name and test file...

File: Makefile [hunk 2]
```diff
@@ -1,16 +1,23 @@
 all: build
 
+export CXX := clang++
+export GTEST_COLOR := on
```
Comment (megabyde) [inline Makefile]: Hopefully, this makes it more explicit

**Review by klimkin — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (klimkin) [reply Makefile]: This template uses Catch2, not GTest.

**Comment by klimkin** [comment]
On a separate note, the build of Catch2 tests is tens of times slower than GTest. I am disappointed.

**Pull request merged**